### PR TITLE
audit fixes: A 03: aescrypt - ant-contrib

### PR DESCRIPTION
**Pull request by apjanke**
audit fixes for: aescrypt-packetizer afio afsctool ant-contrib

(originally included agda, android-ndk, apache-spark, but they were removed due to build/test issues)

### CODE REVIEW

File: Library/Formula/agda.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@ class Agda < Formula
   mirror "https://github.com/agda/agda/archive/2.4.2.3.tar.gz"
   sha256 "bc6def45e32498f51863d67acfbe048c039d630c6a36761ed27e99a5f68d7b27"
   revision 2
+  head "https://github.com/agda/agda.git", :branch => "master"
```
Comment (DomT4) [inline Library/Formula/agda.rb]: `master` never needs specifying on `git` branches. We assume that's the default & don't need pointing unless it isn't.

File: Library/Formula/agda.rb [hunk 2]
```diff
@@ -17,7 +18,11 @@ class Agda < Formula
     sha256 "d0e17808ee896cdedc499b010b67961dc3db3c430c501faef16c4b4482238447" => :mountain_lion
   end
 
-  head "https://github.com/agda/agda.git", :branch => "master"
+  head do
```
Comment (DomT4) [inline Library/Formula/agda.rb]: And you're right about the issue you filed this morning; `head` lines should be merged into `head do` blocks when there's additional resources/deps/etc.

Comment (apjanke) [inline Library/Formula/agda.rb]: Corrected in new commit.

**Comment by apjanke** [comment]
`agda` failed. Looks like it needs `ghc` for its test, but the formula has `depends_on "ghc" => :build`.

If the test ends up using `ghc`, shouldn't that be a normal runtime dependency, too, not just a build-time dependency? This `agda` program is itself a compiler so it makes sense it could need a compiler at run time.

```
/usr/local/Cellar/agda/2.4.2.3_2/bin/agda
-c
--no-main
--safe
/private/tmp/agda20151005-15986-41a7va/simple-test.agda

Checking simple-test (/private/tmp/agda20151005-15986-41a7va/simple-test.agda).

Finished simple-test.

Compiling Agda.Primitive in /usr/local/Cellar/agda/2.4.2.3_2/share/x86_64-osx-ghc-7.10.2/Agda-2.4.2.3/lib/prim/Agda/Primitive.agdai to /private/tmp/agda20151005-15986-41a7va/MAlonzo/Code/Agda/Primitive.hs

Compiling simple-test in /private/tmp/agda20151005-15986-41a7va/simple-test.agdai to /private/tmp/agda20151005-15986-41a7va/MAlonzo/Code/QsimpleZ45Ztest.hs

Calling: ghc -O -Werror -i/private/tmp/agda20151005-15986-41a7va /private/tmp/agda20151005-15986-41a7va/MAlonzo/Code/QsimpleZ45Ztest.hs --make -fwarn-incomplete-patterns -fno-warn-overlapping-patterns

ghc: createProcess: runInteractiveProcess: exec: does not exist (No

such file or directory)

Error: agda: failed

Failed executing: /usr/local/Cellar/agda/2.4.2.3_2/bin/agda -c --no-main --safe /private/tmp/agda20151005-15986-41a7va/simple-test.agda
```

**Comment by DomT4** [comment]
We've been told before by `ghc` people that whilst the current test seems to require `ghc`, runtime `agda` doesn't generally. We should rewrite that test rather than enforce the `ghc` non-build dep.

**Comment by apjanke** [comment]
Maybe we should just remove the `-c` test.

```
system bin/"agda", "-c", "--no-main", "--safe", test_file_path
```
Per `agda --help`, that `-c` is a synonym for ` --compile    compile program using the MAlonzo backend (experimental)`, and [this little wiki entry](http://wiki.portal.chalmers.se/agda/pmwiki.php?n=Docs.MAlonzo) says that means "compile to Haskell and an executable", so that's a functional area that would need `ghc`.

> MAlonzo
>
>Usage
>
> agda --compile --malonzodir <DIR> --ghc-flag “<GHC FLAGS>” <FILE>.agda
>
>Compiles <FILE>.agda and hereditarily imported agda files to Haskell and to an executable. 

Maybe remove that test entirely, or conditionalize it on `ghc` being installed? The other tests still pass in the absence of `ghc`.

**Comment by apjanke** [comment]
I'm pulling the `agda` changes out of this PR so these other fixes can go through, and will re-submit them as a separate single-formula PR. The `agda` build is big enough it's unwieldy to lump in with other formulae.

**Comment by apjanke** [comment]
Last failure:

```
Error Message

failed: brew audit android-ndk

Stacktrace

        Error: 1 problem in 1 formula
android-ndk:
 * object files were linked against system openssl
These object files were linked against the deprecated system OpenSSL.
Adding `depends_on "openssl"` to the formula may help.
  /usr/local/Cellar/android-ndk/r10e/prebuilt/darwin-x86_64/lib/python2.7/lib-dynload/_hashlib.so
  /usr/local/Cellar/android-ndk/r10e/prebuilt/darwin-x86_64/lib/python2.7/lib-dynload/_ssl.so
```

**Comment by DomT4** [comment]
Just waiting for the CI to do this PR again as it decided not to build bottles last time. It might decide that again here as well as `android-ndk` is hilariously large an install and tends to time out the bottle builds.

**Comment by DomT4** [comment]
But you can ignore that audit message. We can't do anything about it, sadly.

**Comment by apjanke** [comment]
I'll break out android-ndk to a separate PR since it's unwieldy.

Should a comment be added to its formula about being okay with the system OpenSSL warning? Or a declaration that would let `brew audit` know to overlook it for this particular formula?

**Comment by DomT4** [comment]
> Should a comment be added to its formula about being okay with the system OpenSSL warning? Or a declaration that would let brew audit know to overlook it for this particular formula?

Leave it as is. I'm perfectly happy with OpenSSL warnings being annoying because they are things that should be fixed, and upstreams should be nagged to make it possible to fix them, especially on a platform where the default OpenSSL is archaic and extremely cipher limited.

**Comment by apjanke** [comment]
I also removed apache-spark from this batch because it's such a big download: 268M, and the URL for it uses a round-robin mirror selection, and some of them are slow. Download can easily take over 10 minutes and trigger the CI's inactivity timeout during the `brew fetch` stage.

**Comment by xu-cheng** [comment]
Rebase is required.

**Comment by apjanke** [comment]
Rebased. Someone else already took care of `afio`, so I removed it from this PR.

**Comment by apjanke** [comment]
Test-bot error: Download failure for aescrypt-packetizer. Maybe transient error; worked fine for me a little while ago. Let's re-test in a while.

**Comment by apjanke** [comment]
Pulling aescrypt-packetizer out to a separate PR so the download failure doesn't bork the rest of this.

I'm really starting to see the wisdom of "one formula per commit", even for little audit/style changes like this.

**Comment by DomT4** [comment]
> I'm really starting to see the wisdom of "one formula per commit", even for little audit/style changes like this.

Heh. It's certainly easier to drop one commit from a rebase than having to manually go in and pull bits out, that's for sure.

Merged in https://github.com/Homebrew/homebrew/commit/fbb2682f172bee512aa82452ae475a7141ad8c43. Thanks again @apjanke :smiley_cat:.

**Comment by apjanke** [comment]
On the upside, I'm getting good `git` practice here! :)

**Comment by DomT4** [comment]
Before I started using Homebrew I used `git` for almost nothing. Was actually a fan of Subversion for a fair while. Since then, git everywhere :smile_cat:.

**Comment by apjanke** [comment]
Git is clearly the winner in this generation of VCS, why fight it? I'm still a fan of Subversion, and they're different animals; hope there's room for both in this world.

**Pull request closed without merging**